packages/shadow-dom/src/findInsertionPoint.ts

```diff
@@ -29,10 +30,14 @@ export function findInsertionPoint(
     const styleElement = styleSheets[i];
 
     if (comparer(styleElement) > 0) {
-      return styleSheets[i + 1] ?? null;
+      return styleSheets[i + 1] ?? insertionPoint ?? null;
```

Comment:
It would be good for this to be consistent with vanilla griffel - the `insertionPoint` is where insertion starts, so all stylesheets should be inserted after th `insertionPoint` but before the insertionPoint's next sibling.

It might make the `insertBefore` handling a bit more tedious but users can expect it to function the same way